src/test/java/histogram/CashierTransactionHistogramTest.java

```diff
@@ -1,36 +1,78 @@
 package histogram;
 
-import domain.POSDetail;
 import org.junit.Before;
 import org.junit.Test;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import utility.ParsingUtility;
 
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Paths;
+
+
 import java.util.HashMap;
-import java.util.List;
+
 import java.util.Map;
 
+import static junit.framework.TestCase.assertEquals;
 import static junit.framework.TestCase.assertTrue;
 
 
 public class CashierTransactionHistogramTest{
 
-    String inputFile = "/Users/aap1018/Documents/workspace/minorityReportCopy/minorityreport/trendsDataDao/java/transactionCompare/src/main/resources/posDetails_03_2016.csv";
+    String inputFile = "/Users/aap1018/Documents/workspace/javaProjectTemplate/src/main/resources/input/txTest.csv";
     static Logger logger = LoggerFactory.getLogger(CashierTransactionHistogramTest.class);
-    Map<String, Map<String, Double>> targetHistogram = new HashMap<>();
+    Map<String, Map<String, Double>> actualHistogram = new HashMap<>();
+    Map<String, Map<String, Double>> expectedHistogram = new HashMap<>();
+    CashierTransactionHistogram cashierHistogram;
+
     @Before
-    public void setUp() throws IOException {
-        CashierTransactionHistogram cashierHistogram = new CashierTransactionHistogram(inputFile);
+    public void setUp() {
+        cashierHistogram = new CashierTransactionHistogram(inputFile);
+        actualHistogram = cashierHistogram.getCashierHourOfDayHistogram();
+
 
 
     }
 
     @Test
-    public void cashierTransactionHistogramTest
+    public void targetHistogramTest(){
+        Map<String,Double> txMap1 = new HashMap<>();
+        Map<String,Double> txMap2 = new HashMap<>();
+        Map<String,Double> txMap3 = new HashMap<>();
+        txMap1.put("3_AgeVerifyDOB", 1.0);
+        expectedHistogram.put("3_5662593", txMap1);
+        txMap2.put("2_Entry", 1.0);
+        expectedHistogram.put("2_5662593", txMap2);
+        txMap3.put("1_SignInOut",2.0);
+        expectedHistogram.put("1_5662593",txMap3);
+
+        assertTrue("Error: Unexpected data", (actualHistogram.keySet().size()==expectedHistogram.keySet().size()));
+
+        for(String cashier: expectedHistogram.keySet()){
+            assertTrue("Error: Unexpected data", (actualHistogram.containsKey(cashier)));
+        }
+
+        for(String cashier: expectedHistogram.keySet()){
+            Map<String,Double> expectedMap = expectedHistogram.get(cashier);
+            Map<String, Double> actualMap = actualHistogram.get(cashier);
+
+            assertTrue("Error: Unexpected data", (expectedMap.keySet().size()==actualMap.keySet().size()));
+
+            for(String t: expectedMap.keySet()){
+                assertTrue("Error: Unexpected data", (actualMap.containsKey(t)));
```

Comment:
Not sur what the message is expected to convey - because we are assertingTrue

Reply:
Conditions that first check equality of hash map keyset lengths, then whether the keys in expected exist in actual. An assertion is raised if the conditions evaluate to false